File: src/main/java/sage/web/context/Json.java
package sage.web.context;

import java.io.IOException;

import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.JavaType;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.type.TypeFactory;

public abstract class Json {
  private static final ObjectMapper om = new ObjectMapper();

  public static String json(Object object) {
    try {
      return om.writeValueAsString(object);
    }
    catch (JsonProcessingException e) {
      throw new IllegalArgumentException(e);
    }
  }
  
  public static <T> T object(String json, Class<T> type) {
    try {
      return om.readValue(json, type);
    } catch (IOException e) {
      throw new IllegalArgumentException(e);
    }
  }

  public static <T> T object(String json, JavaType jt) {
    try {
      return om.readValue(json, jt);
    } catch (IOException e) {
      throw new IllegalArgumentException(e);
    }
  }

  public static TypeFactory typeFactory() {
    return om.getTypeFactory();
  }
}

File: src/main/kotlin/sage/util/Json.kt
package sage.util

import com.fasterxml.jackson.core.JsonProcessingException
import com.fasterxml.jackson.databind.DeserializationFeature
import com.fasterxml.jackson.databind.JavaType
import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.databind.type.TypeFactory
import java.io.IOException

object Json {
  private val om = ObjectMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)

  @JvmStatic fun json(`object`: Any): String {
    try {
      return om.writeValueAsString(`object`)
    } catch (e: JsonProcessingException) {
      throw IllegalArgumentException(e)
    }

  }

  fun <T> `object`(json: String, type: Class<T>): T {
    try {
      return om.readValue(json, type)
    } catch (e: IOException) {
      throw IllegalArgumentException(e)
    }

  }

  fun <T> `object`(json: String, jt: JavaType): T {
    try {
      return om.readValue<T>(json, jt)
    } catch (e: IOException) {
      throw IllegalArgumentException(e)
    }

  }

  fun typeFactory(): TypeFactory {
    return om.typeFactory
  }
}
